carts/views.py

```diff
@@ -1,3 +1,58 @@
-from django.shortcuts import render
+import json, re, bcrypt, jwt
 
-# Create your views here.
+from django.views           import View
+from django.http            import JsonResponse
+from django.db              import transaction
+from django.db.models       import Q
+
+from products.models        import Product,ProductOption, Size, Color, CustomImage, CustomProduct, FontColor, FontStyle
+from carts.models           import Cart
+from users.utils            import login_decorator
+
+class CustomProductView(View):
+    @login_decorator
+    @transaction.atomic
+    def post(self, request, product_id):
+        try:
+            user       = request.user
+            data       = json.loads(request.body)
+
+            product = ProductOption.objects.get(
+                product_id = product_id, 
+                size       = data['size_id'],
+                color      = data['color_id'],
+            )
+
+            if not product.stock:
+                return JsonResponse({'MESSAGE':'OUT_OF_STOCK'}, status = 400)
+        
+            product.stock = product.stock-1
+            product.sales = product.sales+1
+            product.save()
+
+            q = Q(user_id = user.id) & Q(product_id = product_id)
+            
+            customproduct, custom_is_created = CustomProduct.objects.get_or_create(
+                user_id         = user.id,
+                product_id      = product_id,
+                font_color_id   = data.get('font_color_id'),
+                font_style_id   = data.get('font_style_id'),
+                custom_image_id = data.get('custom_image_id'),
+                custom_text     = data.get('custom_text'),
+                coordinate_x    = data.get('coordinate_x'),
+                coordinate_y    = data.get('coordinate_y'),
+            )
+
+            cart_product, cart_is_created = Cart.objects.get_or_create(user_id = user.id, custom_product_id = customproduct.id, defaults={'quantity' : 1 })
+            if not cart_is_created:
+                quantity              = cart_product.quantity
+                quantity              = quantity + 1
+                cart_product.quantity = quantity
+                cart_product.save()
```

Comment:
```suggestion
            cart, cart_is_created = Cart.objects.get_or_create(
                user_id           = user.id, 
                custom_product_id = customproduct.id, 
                defaults           = {'quantity' : 1 }
            )
            
            if not cart_is_created:
                cart_product.quantity += 1
                cart_product.save()
```
cart에 추가하는 로직을 다음과 같이 바꿔도 될 것같은데 한 번 확인해 보세요.

Reply:
넵 !